Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -263,85 +398,60 @@ TextBox::TextBox(QString current, QWidget* parent)
 TextBox::TextBox(std::shared_ptr<TextModel> current, QWidget* parent)
     : QWidget(parent),
       m_current(std::move(current)),
-      m_submission(m_current->get()),
+      m_submission(std::make_shared<LocalTextModel>(m_current->get())),
+      m_read_only(std::make_shared<LocalBooleanModel>(false)),
```

Comment:
I can maybe sympathize with the submission being a model, but read-only should not be a model. Models introduce a form of non-local reasoning along with all the problems that go with it (reentrancy bugs being the big one). If something is a model then I have to go over all connections to it to verify that there's no reentrancy issues, as opposed to if something is a local field then all of its uses are isolated.